src/OrleansAzureUtils/Storage/Exceptions.cs

```diff
@@ -48,7 +48,7 @@ public class TableStorageUpdateConditionNotSatisfiedException : InconsistentStat
         /// <summary>
         /// Id of grain
         /// </summary>
-        public GrainReference GrainId { get; }
+        public string GrainId { get; }
```

Comment:
It's because GrainReferences internally need a reference to `IRuntimeClient`, which BinaryFormatterSerializer can't provide. We need that info when calling `FromKeyString` below.